executor/analyze.go

```diff
@@ -264,19 +265,45 @@ func analyzeColumnsPushdown(colExec *AnalyzeColumnsExec) statistics.AnalyzeResul
 
 // AnalyzeColumnsExec represents Analyze columns push down executor.
 type AnalyzeColumnsExec struct {
-	ctx         context.Context
-	tblInfo     *model.TableInfo
-	colsInfo    []*model.ColumnInfo
-	pkInfo      *model.ColumnInfo
-	concurrency int
-	priority    int
-	keepOrder   bool
-	analyzePB   *tipb.AnalyzeReq
-	result      distsql.SelectResult
+	ctx           context.Context
+	tblInfo       *model.TableInfo
+	colsInfo      []*model.ColumnInfo
+	pkInfo        *model.ColumnInfo
+	concurrency   int
+	priority      int
+	keepOrder     bool
+	analyzePB     *tipb.AnalyzeReq
+	resultHandler *tableResultHandler
 }
 
 func (e *AnalyzeColumnsExec) open() error {
-	ranges := ranger.FullIntNewRange()
+	var ranges []*ranger.NewRange
```

Comment:
This part of code is repeated with `TableReader`. How about using `ranges`, `keepOrder` to initialize `resultHandler` ?

Reply:
the `kvReq` they build is different.